cfme/infrastructure/deployment_roles.py

```diff
@@ -1,67 +1,266 @@
+    reset_button = Button('Reset')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        """Is this page currently being displayed"""
+        return (
+            self.in_dep_role and
+            self.breadcrumb.active_location == 'Tag Assignment')
+
+
+class DeploymentRoleManagePoliciesView(DeploymentRoleView):
+    """Deployment role Manage Policies view."""
+    breadcrumb = BreadCrumb()
+    policies = BootstrapTreeview("protectbox")
+    save_button = Button("Save")
+    reset_button = Button("Reset")
+    cancel_button = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        """Is this page currently displayed"""
+        return (
+            self.in_dep_role and
+            (self.breadcrumb.active_location == "'Cluster / Deployment Role' Policy Assignment" or
+             self.breadcrumb.active_location == "'Deployment Role' Policy Assignment")
+        )
+
+
+class DeploymentRoles(Navigatable):
     """ Model of an infrastructure deployment roles in cfme
 
     Args:
         name: Name of the role.
         provider: provider this role is attached to
             (deployment roles available only for Openstack!).
     """
-    pretty_attrs = ['name', 'provider']
 
     def __init__(self, name, provider, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
-        if not isinstance(provider, OpenstackInfraProvider):
-            raise NotImplementedError('Deployment roles available only '
-                                      'for Openstack provider')
         self.name = name
         self.provider = provider
 
-    def get_detail(self, *ident):
-        """ Gets details from the details InfoBlock
-        Args:
-            *ident: An InfoBlock title, followed by the Key name,
-                e.g. "Relationships", "All VMs"
-        Returns: A string representing the contents of the InfoBlock's value.
-        """
-        navigate_to(self, 'Details')
-        return details_page.infoblock.text(*ident)
+        if not provider.one_of(OpenstackInfraProvider):
+            raise NotImplementedError('Deployment roles available only '
+                                      'for Openstack provider')
+        Navigatable.__init__(self, appliance=appliance)
+
+    def delete(self, cancel=False):
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove item',
+                                               handle_alert=not cancel)
 
-    def delete(self):
-        navigate_to(self, 'Details')
-        menu_item = version.pick({version.LOWEST: 'Remove from the VMDB',
-                                  '5.7': 'Remove item'})
-        cfg_btn(menu_item, invokes_alert=True)
-        sel.handle_alert(wait=60)
+        if not cancel:
+            view = self.create_view(DeploymentRoleAllView)
+            assert view.is_displayed
+            view.flash.assert_success_message("The selected Clusters / "
```

Comment:
is there a bug about that ?

Reply:
No.

Reply:
Please create a BZ if it doesn't exist yet.